### PR TITLE
Readme + Code commit

### PR DESCRIPTION
**Pull request by Sm1Ling**
Added project description in README.md

### CODE REVIEW

**Comment by Sm1Ling** [comment]
modified readme

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Курс "Code as Data"
+## Аннотация
+> Текущий ридми содержит описание того что предстоит сделать. Автор репозитория Арифханов Азаматхан, БПИ 198
```
Comment (EgorBu) [inline README.md]:
```suggestion
> Здесь содержится описание того, что предстоит сделать за время исполнения проекта. Автор репозитория Арифханов Азаматхан, БПИ 198
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Планируемые результаты обучения
+* Понимание важных метрик, используемых на практике для определения похожести кода и разработчиков
+* Углубленное понимание принципов работы GIT
+* Практика работы с CLI (Command-line interface) гита
```
Comment (EgorBu) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Курс "Code as Data"
+## Аннотация
+> Текущий ридми содержит описание того что предстоит сделать. Автор репозитория Арифханов Азаматхан, БПИ 198
+
+## Планируемые результаты обучения
+* Понимание важных метрик, используемых на практике для определения похожести кода и разработчиков
+* Углубленное понимание принципов работы GIT
```
Comment (EgorBu) [inline README.md]:
```suggestion
* Углубленное понимание принципов работы git и использование git для получения желаемой информации из репозиториев
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+* Практика работы с CLI (Command-line interface) гита
+* Практика работы с утилитами для парсинга репозиториев, извелечения информации о коммитах
```
Comment (EgorBu) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* Составление датасетов для дальнейшей работы с ними
```
Comment (EgorBu) [inline README.md:10]: ~ тоже самое для enry/tree-sitter как для git'а

File: README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		<p> - 5.1.1. Используем tree-sitter для формирования AST кода
+		<p> - 5.1.2. ...
+	<p>5.2. Составляем вектор зависимостей и имен (имена переменных, функций, импортов)
+		<p> - 5.2.1. ...
+6. Метрики похожести
+	<p>6.1. Подбор метрик похожести  (Косинусная дистанция, L2/L1 мера)
+	<p>6.2. ...
+
+### Литература
+* [Курочка-Ряба](https://github.com/Sm1Ling/Ryaba)
```
Comment (EgorBu) [inline README.md]:
```suggestion
```

**Review by EgorBu — CHANGES_REQUESTED** [review]
Let's squash commits & fix comments - and we can merge)

**Review by EgorBu — APPROVED** [review]
LGTM! Let's squash all commits and than merge

File: git_repo_extract/commits_info.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import json
+
+from dulwich.diff_tree import TreeChange
```
Comment (EgorBu) [inline git_repo_extract/commits_info.py:5]: please fix  order of imports

File: git_repo_extract/commits_info.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+def get_diffs_num(repos: Repo, sha1: bytes, sha2: bytes) -> Tuple[int, int]:
+    """
+    A method that gives blob differences
+    Args:
+        :param repos: source repository
+        :param sha1: old blob's sha
+        :param sha2: new blob's sha
+    Returns:
+        :return [Added, Deleted]
```
Comment (EgorBu) [inline git_repo_extract/commits_info.py]:
```suggestion
        :return (Added, Deleted)
```

File: git_repo_extract/commits_info.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        :param repo: source repository
+        :param limit: limit of entities to check. Useful for pipeline check
+      Returns:
+        :return Iterator of dicts
+    """
+    repo_url = (repo.get_config().get((b"remote", b"origin"), b"url")).decode()
+    i = 0
+    for walk in repo.get_walker():
+        name = walk.commit.author.decode()
+        mail = name[name.find('<') + 1:-1]
```
Comment (EgorBu) [inline git_repo_extract/commits_info.py]: let's use double quotes everywhere

File: git_repo_extract/star_track.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+from tqdm import tqdm
+from commits_info import write_down_content
+
+from github import Github, RateLimitExceededException
+from typing import Dict, Any, List
+
+
+def get_stargazer_info(repo_name: str, github_key: str,
+                       max_user_stars: int = 300,
+                       per_page: int = 80) -> Any[List[Dict], Dict]:
```
Comment (EgorBu) [inline git_repo_extract/star_track.py]: docstring doesn't match return type

**Review by EgorBu — CHANGES_REQUESTED** [review]
Almost there! Good job

**Review by EgorBu — APPROVED** [review]
Awesome!

**Pull request merged**